### PR TITLE
preserve chain and draw dims in posterior predictive

### PR DESCRIPTION
**Pull request by OriolAbril**
+ [x] what are the (breaking) changes that this PR makes? None that I can think of
+ [x] important background, or details about the implementation
  - This is mostly relevant when thinning the posterior to generate posterior predictive samples
    for only a subset of it (if pp sampling is expensive for example), but can also
    be an issue if someone uses non standard coordinate values.
  - It depends on https://github.com/arviz-devs/arviz/pull/2001 and an ArviZ release before the
    specific test can pass. Should I use a skipif to request arviz>0.11.4 so it can be merged
    before that? (not sure when next ArviZ release will be)
+ [x] are the changes—especially new features—covered by tests and docstrings?
+ [x] [linting/style checks have been run](https://docs.pymc.io/en/latest/contributing/python_style.html)
+ [x] [consider adding/updating relevant example notebooks](https://github.com/pymc-devs/pymc-examples): Updating the golf putting notebook that samples posterior predictive on a thinned posterior only.
+ [ ] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

### CODE REVIEW

**Comment by OriolAbril** [comment]
This should be ready to merge now that arviz 0.12 is out. The specific tests that need arviz>=0.12 are passing like all other tests and only pre-commit is failing due to the black issue mentioned in slack.

**Comment by OriolAbril** [comment]
@michaelosthege I rebased after merging the fix ci PR but mypy seems to be broken again

File: pymc/sampling.py [hunk 1]
```diff
@@ -1620,10 +1620,20 @@ def sample_posterior_predictive(
...
     nchain: int
+    if idata_kwargs is None:
+        idata_kwargs = {}
+    else:
+        idata_kwargs = idata_kwargs.copy()
+    if "coords" not in idata_kwargs:
+        idata_kwargs["coords"] = {}
     if isinstance(trace, InferenceData):
+        idata_kwargs["coords"].setdefault("draw", trace.posterior.draw)
+        idata_kwargs["coords"].setdefault("chain", trace.posterior.chain)
```
Comment (michaelosthege) [inline pymc/sampling.py]: Mypy isn't broken - you just introduced two lines that according to the type information from ArviZ don't work.

Replace `trace.posterior` with `trace["posterior"]` in the two lines above

Comment (OriolAbril) [reply pymc/sampling.py]: Oohh, sorry. Now I see what Ravin meant in slack. These two lines prompted over 350 lines of mypy warnings and errors that start with aesaraf related things

**Review by michaelosthege — APPROVED** [review]

**Pull request merged**